packages/shared/components/Search/SearchPanel.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Text, Flex } from 'design';
+import { StyledPanel } from 'design/DataTable';
+import { PageIndicatorText } from 'design/DataTable/Pager/Pager';
+import InputSearch from 'design/DataTable/InputSearch';
+import { AgentFilter } from 'teleport/services/agents';
+import Toggle from 'teleport/components/Toggle';
+import Tooltip from 'teleport/components/ServersideSearchPanel/Tooltip';
+
+import { PredicateDoc } from './PredicateDoc';
+
+export function SearchPanel({
+  updateQuery,
+  updateSearch,
+  pageCount,
+  filter,
+  showSearchBar,
+  disableSearch,
+}: {
+  updateQuery(s: string): void;
+  updateSearch(s: string): void;
+  pageCount: { from: number; to: number; total: number };
+  filter: AgentFilter;
+  showSearchBar: boolean;
+  disableSearch: boolean;
+}) {
+  const [query, setQuery] = useState(filter.search || filter.query || '');
+  const [isAdvancedSearch, setIsAdvancedSearch] = useState(!!filter.query);
+
+  useEffect(() => {
+    setIsAdvancedSearch(!!filter.query);
+    setQuery(filter.search || filter.query || '');
+  }, [filter]);
+
+  function onToggle() {
+    setQuery('');
```

Comment:
Do we need to reset the query here? The Web UI doesn't do this, if I start typing an advanced query only to realize I forgot to turn the toggle, it's pretty disruptive for the app to clear the whole query.

Reply:
Updated